clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -540,6 +540,14 @@ public synchronized FetchPosition position(TopicPartition tp) {
         return assignedState(tp).position;
     }
 
+    public synchronized FetchPosition positionOrNull(TopicPartition tp) {
```

Comment:
In all new methods, I try to avoid the `IllegalStateException` by verifying whether the partition is assigned or not. In most methods the return value signals to the caller if the partition is unassigned.